### DISCUSSION TITLE
Producer memory leaks

### DISCUSSION THREAD

**Discussion by ghost** [edenhill/librdkafka#3989] [Q&A]
After creating a producer with rd_kafka_new(RD_KAFKA_PRODUCER, producerConfig, errorBuffer, sizeof(errorBuffer));
rd_kafka_destroy doesn't free all the memory allocated for the producer object. After multiple publishing there are significant memory leaks. Please advise how to prevent it or if there has been a code fix for this problem.

**Comment by pranavrth** [comment]
If you feel that this is an issue with librdkafka, please create an issue for the same using template https://github.com/edenhill/librdkafka/issues/new
